File: number-of-different-integers-in-a-string/number-of-different-integers-in-a-string.java
class Solution {
    public int numDifferentIntegers(String word) {
        
        HashSet<String> set = new HashSet<>();
        
        for(int i = 0; i < word.length(); i++){
            
            if(Character.isDigit(word.charAt(i))){
                int j = i;
                String temp = "";
                while(j < word.length() && Character.isDigit(word.charAt(j))){
                    if(temp.equals("") && word.charAt(j) == '0'){
                        j++;
                        continue;
                    }
                    temp += Character.toString(word.charAt(j));
                    j++;
                }
                i = j;
                set.add(temp);
            }
        }
        
        return set.size();
    }
}

File: number-of-different-integers-in-a-string/number-of-different-integers-in-a-string.cpp
class Solution {
public:
    int numDifferentIntegers(string word) {
        
        set<string> res;
        for(int i = 0; i < word.length(); i++){
            if(isdigit(word[i])){
                int j = i;
                string temp = "";
                while(j < word.length() && isdigit(word[j])){
                    if(temp == "" && word[j] == '0'){
                        j++;
                        continue;
                    }
                    temp += word[j];
                    j++;
                }
                res.insert(temp);
                i = j;
            }
        }
        
        return res.size();
    }
};